### ISSUE TITLE
restore aspnetcore source code when exec restore.sh script faild.

### ISSUE THREAD

**Issue by kssoftwangwg** [dotnet/aspnetcore#29802]
### Describe the bug
I want build aspnetcore v3.1.11 by aspnetcore source code on mac os , git checkout v3.1.11 tag build it. I modified some source files,They are:
(1).global.json
``` json
{
  "sdk": {
    "version": "3.1.405" // Modified here
  },
  "tools": {
    "dotnet": "3.1.405",  // Modified here
    "runtimes": {
      "dotnet/x64": [
        "$(MicrosoftNETCoreAppInternalPackageVersion)"
      ],
      "dotnet/x86": [
        "$(MicrosoftNETCoreAppInternalPackageVersion)"
      ]
    },
    "Git": "2.22.0",
    "jdk": "11.0.3",
    "vs": {
      "version": "16.3",
      "components": [
        "Microsoft.VisualStudio.Component.VC.ATL",
        "Microsoft.VisualStudio.Component.VC.Tools.x86.x64",
        "Microsoft.VisualStudio.Component.Windows10SDK.17134"
      ]
    }
  },
  "msbuild-sdks": {
    "Yarn.MSBuild": "1.15.2",
    "Microsoft.DotNet.Arcade.Sdk": "1.0.0-beta.20509.7",
    "Microsoft.DotNet.Helix.Sdk": "2.0.0-beta.20509.7"
  }
}
```
(2). eng folder Versions.props MicrosoftNETCoreAppInternalPackageVersion value modified：
value : `MicrosoftNETCoreAppInternalPackageVersion>3.1.11</MicrosoftNETCoreAppInternalPackageVersion`  Because I want to compile it is this version, so I modified here.

but when i execute restore.sh command to restore csproj package is referenced in Src folder，console output errors:
``` text
_/Users/miracle/Desktop/dotnetcore/aspnetcore/src/Middleware/HealthChecks/test/testassets/HealthChecksSample/HealthChecksSample.csproj : error NU1605: Detected package downgrade: Microsoft.Extensions.DependencyInjection.Abstractions from 3.1.11 to 3.1.10. Reference the package directly from the project to select a different version.  [/Users/miracle/Desktop/dotnetcore/aspnetcore/.dotnet/sdk/3.1.405/NuGet.targets]
and more package downgrade errors._
```

**_I don't know how I should modify to successfully restore project dependencies without reporting errors._**，Please help me resolve these errors, thank you very much！

If you want to see more console output, please go to：https://gist.github.com/kssoftwangwg/fb3b60d11809258b89cd01d3bc946524 . filename is: **aspnetcore exec restore.sh failed console.log on mac os  **  View details.

### Further technical details
- v3.1.11
- 运行时环境:
 OS Name:     Mac OS X
 OS Version:  10.15
 OS Platform: Darwin
 RID:         osx.10.15-x64
 Base Path:   /usr/local/share/dotnet/sdk/5.0.102/

Host (useful for support):
  Version: 5.0.2
  Commit:  cb5f173b96

.NET SDKs installed:
  3.1.302 [/usr/local/share/dotnet/sdk]
  3.1.401 [/usr/local/share/dotnet/sdk]
  3.1.402 [/usr/local/share/dotnet/sdk]
  3.1.403 [/usr/local/share/dotnet/sdk]
  3.1.404 [/usr/local/share/dotnet/sdk]
  3.1.405 [/usr/local/share/dotnet/sdk]
  5.0.100 [/usr/local/share/dotnet/sdk]
  5.0.101 [/usr/local/share/dotnet/sdk]
  5.0.102 [/usr/local/share/dotnet/sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.6 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.7 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.8 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.9 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.10 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 3.1.11 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.0 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.1 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.2 [/usr/local/share/dotnet/shared/Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 2.1.20 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.21 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.22 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 2.1.23 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.6 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.7 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.8 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.9 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.10 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 3.1.11 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.0 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.1 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.2 [/usr/local/share/dotnet/shared/Microsoft.NETCore.App]

To install additional .NET runtimes or SDKs
- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version

**Comment by dougbu** [comment]
@kssoftwangwg thank-you for your feedback. We're closing this issue as we think this is fixed in our working code base.

The v3.1.11 tag points to an incorrect commit -- 1b1646a5a6b6 was before we rebranded for 3.1.11. I believe it should point to 9699b939f94b. I checked more recent v3.1.x tags and didn't see a similar problem.

@leecow @mmitche @wtgodbe were we automating tag creation when 3.1.11 went out❔ My concern the possibility of other incorrect v3.1.11 tags or even a broader problem. Either way, should I manually update this particular tag in dotnet/aspnetcore❔

**Comment by dougbu** [comment]
Actually, I'll close this after we fix the v3.1.11 tag.

**Comment by dougbu** [comment]
Very bizarre: After a fair amount of "all is well" signals as I trolled though https://github.com/dotnet/{blah}/releases/tag/v3.1.11 links and the manifest.json file for 3.1.11 / 3.1.111, I hit https://github.com/dotnet/efcore/releases/tag/v3.1.11 and that looks like it points to a 5.0.x commit instead of dotnet/efcore@b64867f633c6b4688a97750724728ec914e26d3e. Releasing the aspnetcore targeting pack doesn't explain that one at all. Any ideas❔

**Comment by dougbu** [comment]
Fixed the v3.1.11 tag in dotnet/efcore and am declaring success (though I'm confused how the automation could have messed up that tag).

Thanks again for the report @kssoftwangwg❕

**Comment by wtgodbe** [comment]
No idea how a 5.0 tag would've gotten in to the 3.1 manifest (or how else 3.1.11 could've been tagged with a 5.0 commit) - @rbhanda is there some kind of audit trail for this?
